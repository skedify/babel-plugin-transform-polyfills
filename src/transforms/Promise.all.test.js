import compile from '../../test/utils/compile'

describe('Promise.all', () => {
  it('should replace Promise.all with a polyfill', () => {
    const result = compile(
      `
      const c = Promise.all([])
    `
    )

    expect(result).toMatchSnapshot()
  })

  it('should not import polyfill more than once', () => {
    const result = compile(
      `
      const e = Promise.all([Promise.all([1, 2])])
    `
    )

    expect(result).toMatchSnapshot()
  })

  it('should work when part of a call chain', () => {
    const result = compile(
      `
      const c = Object.keys(Promise.all([]))
    `
    )

    expect(result).toMatchSnapshot()
  })
})
