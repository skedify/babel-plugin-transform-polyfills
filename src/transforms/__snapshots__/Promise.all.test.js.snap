// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Promise.all should not import polyfill more than once 1`] = `
"import _Promise_all from \\"babel-plugin-transform-polyfills/polyfills/Promise/all\\";

const e = _Promise_all([_Promise_all([1, 2])]);"
`;

exports[`Promise.all should replace Promise.all with a polyfill 1`] = `
"import _Promise_all from \\"babel-plugin-transform-polyfills/polyfills/Promise/all\\";

const c = _Promise_all([]);"
`;

exports[`Promise.all should work when part of a call chain 1`] = `
"import _Promise_all from \\"babel-plugin-transform-polyfills/polyfills/Promise/all\\";

const c = Object.keys(_Promise_all([]));"
`;
